### PR TITLE
606-disableTrackLatestRelease

### PR DESCRIPTION
**Pull request by Lucaszw**
Resolves: #606 
Change-type: minor
See: https://github.com/balena-io-projects/staged-releases/blob/master/disable-rolling-release-on-fleet.sh

### CODE REVIEW

**Comment by pimterry** [comment]
Hi @Lucaszw could you tell me a little more about your use case for this?

I think this method was originally omitted because we assumed users would typically want to disable tracking & pin to a specific release, rather than just disabling release tracking in general, and you can already do the former with `application.pinToRelease`. I'd love to know more about what you're doing, and why this extra method is useful to you.

File: lib/models/application.coffee [hunk 1]
```diff
@@ -874,6 +874,62 @@ getApplicationModel = (deps, opts) ->
...
+	# 	...
+	# });
+	#
+	# @example
+	# balena.models.application.disableTrackLatestRelease('MyApp', function(error) {
+	# 	if (error) throw error;
+	# 	...
+	# });
+	###
+	exports.disableTrackLatestRelease = (nameOrId, callback) ->
```
Comment (thgreasi) [inline lib/models/application.coffee]: I think that the name of the method doesn't fully describe what it's doing under the hood.
It sounds like it should just pin the application to its current release but the implementation actually pins it to the latest release instead, possibly resulting the whole fleet to download updates if a successful release was published in between two calls of this method.

**Comment by nazrhom** [comment]
@resin-ci retest

**Comment by Lucaszw** [comment]
We are looking to have the capability to stop a fleet of deployed machines (on different release commits) from automatically updating on new releases. 

With additional consideration, we are more looking to add "disableRollingUpdates" from the python API to the node.js API (see: https://www.balena.io/docs/reference/sdk/python-sdk/#function-disable_rolling_updates-app_id)

I.e. Add this function to the Node.js Library.

https://github.com/balena-io/balena-sdk-python/blob/master/balena/models/application.py
```python

 def disable_rolling_updates(self, app_id):
        """
        Disable Rolling update on application.
        Args:
            name (str): application id.
        Returns:
            OK/error.
        Raises:
            ApplicationNotFound: if application couldn't be found.
        Examples:
            >> > balena.models.application.disable_rolling_updates('106640')
            'OK'
        """

        params = {
            'filter': 'id',
            'eq': app_id
        }
        data = {
            'should_track_latest_release': False
        }

        return self.base_request.request(
            'application', 'PATCH', params=params, data=data,
            endpoint=self.settings.get('pine_endpoint')
        )

```

I will close this PR, and create a new one.

**Comment by thgreasi** [comment]
> a fleet of deployed machines (on different release commits) 

I guess that some of those devices are pinned to specific releases, is that correct?
Changing the app's tracked release only affects those that are not pinned and the pinned ones should be unaffected (regardless of the app's `commit` or `should_track_latest_release`).

That could be done with the current methods by using:
```
const currentReleaseHash = await balena.models.application.getTargetReleaseHash('myApp');
balena.models.application.pinToRelease('myApp', currentReleaseHash);
```
On the other hand of course you can avoid working with the release hash by making the method to be a simple PATCH, like the one you mentioned in #606 (or similar to what the python sdk does).
I like the idea of having a separate `pinToCurrentRelease()` method that normalized the app identification parameter and then does the aforementioned PATCH.

**Comment by Lucaszw** [comment]
> I guess that some of those devices are pinned to specific releases, is that correct?

That is correct.
We have many deployed machines on different releases, and we want every machine under the current app (regardless of the hash) to be "pinned" to the current release on the device. 

I.e. Maybe something more along the lines of ```application.pinAll(appId)``` that pins each device to its current release, not the current the release on the app level.

I.e. functionality that would have the same outcome of the following (js pseudo code)
```javascript
model.device.getAllByApplication().then((devices) => {
   model.devices.forEach((device) => {
     model.device.pinToRelease(device.id, device.commit);
  });
})

```

**Comment by thgreasi** [comment]
@Lucaszw Your last message confused me a bit.
This PR was about a method in the application model and later you mentioned that you were interested at having the python SDK's `disable_rolling_updates()` method in the JS sdk.
On the other hand the provided pseudocode is different from the other implementations discussed earlier and just each individual device of an application (to what it's currently running).

What's the use case of that? Is this for example so that your fleet only downloads updates in specific time frames?
Is this something you need to do often?

**Pull request closed without merging**